src/Symfony/Component/Serializer/Mapping/Factory/CacheClassMetadataFactory.php

```diff
@@ -46,6 +46,7 @@ public function getMetadataFor($value)
         $class = $this->getClass($value);
         // Key cannot contain backslashes according to PSR-6
         $key = strtr($class, '\\', '_');
+        $key = false !== strpbrk($key, '{}()/@:') ? rawurlencode($key) : $key;
```

Comment:
What about replacing the previous line by this instead?
`$key = strtr($class, '\\:', '__');`

Reply:
@nicolas-grekas  Applying your suggestion will result in the following and still fail
```
class.anonymous/symfony/src/Symfony/Component/Serializer/Tests/Mapping/Factory/CacheMetadataFactoryTest.php-69$b7
```

The way it is now it results in 
```
class%40anonymous%00%2Fsymfony%2Fsrc%2FSymfony%2FComponent%2FSerializer%2FTests%2FMapping%2FFactory%2FCacheMetadataFactoryTest.php%3A69%24b7
```

How about we do it similar to the change [here](https://github.com/symfony/symfony/pull/29313/files#diff-3e45cf556c0330623c0167841d303dffR511) and replace both lines with

```
$key = false !== strpbrk($class, '{}()/@:') ? rawurlencode($class) : $class;
```

Reply:
> How about we do it similar to the change here and replace both lines with

works for me

Reply:
The last change works for the serializer component but affects a test case in the FrameworkBundle. The keys used [here](https://github.com/symfony/symfony/blob/v4.4.11/src/Symfony/Bundle/FrameworkBundle/Tests/CacheWarmer/SerializerCacheWarmerTest.php#L40) do not match anymore. I am not really sure about the impact this change has. Should I adjust the test case or revert it back to the first version which did not cause this? :thinking: